src/Exercise04AI.ipynb

```diff
@@ -220,14 +223,16 @@
    "source": [
     "# Test minimax again (now memoized)\n",
     "def test_memoized_minimax(unit_test_player: ChessAI, board: chess.Board):\n",
-    "    unit_test_player.cache_hits = 0  # Reset cache hits counter\n",
+    "    unit_test_player.stats[-1][\"cache_hits\"] = 0  # Reset cache hits counter\n",
     "    mm_evaluation, mm_move = unit_test_player.minimax(board, 2, 1240)\n",
     "    assert mm_evaluation == 355, \"Minimax evaluation does not match expected value!\"\n",
     "    assert mm_move.uci() == \"d4c3\", \"Minimax move does not match expected value!\"\n",
-    "    assert unit_test_player.cache_hits == 1, \"Cache was not used correctly!\"\n",
+    "    assert (\n",
+    "        unit_test_player.stats[-1][\"cache_hits\"] == 1\n",
+    "    ), \"Cache was not used correctly!\"\n",
     "    print(f\"Minimax Evaluation: {mm_evaluation}\")\n",
     "    print(f\"Minimax Move: {mm_move}\")\n",
-    "    print(f\"Cache hits: {unit_test_player.cache_hits}\")"
+    "    print(f\"Cache hits: {unit_test_player.stats[-1]['cache_hits'] }\")"
```

Comment:
Würd ich ebenso um Cache Misses ergänzen

Reply:
Fixed in d4e37ec.